Core/include/Components/ListenerComponent.h

```diff
@@ -0,0 +1,27 @@
+#ifndef LSTNCOMP_H
+#define LSTNCMOP_H
+
+#include <Component.h>
+#include <Managers/AudioManager.h>
+
+#include <fmod.hpp>
+#include <glm.hpp>
+
+class ListenerComponent : public Component {
+public:
+	static ListenerComponent* getInstance(Entity* mEntity, AudioManager* audioManager);
```

Comment:
How would you approach this. Make it not singleton and hope the user doesn't make multiple?

Reply:
Read the rest of the comments first

Reply:
What if a game has splitscreen or co-op, you could do audio channelling or client side listening. 

In industry juniors and co-ops tended to overuse singletons because it ensures one instance but remember- you are introducing a global variable- you need to have an incredibly good reason to do that.

We use EntityManager to get entities accross the entire ECS system, and its significance is global enough to warrant it.
We already have all the faculties to get something with a listener component on it, so why not just use it? We introduce a Singleton here for convenience only -thats bad design.